File: AppListaCursoJava/app/src/main/java/com/santos/valdomiro/applistacursojava/view/MainActivity.java
package com.santos.valdomiro.applistacursojava.view;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import com.santos.valdomiro.applistacursojava.controller.PessoaController;
import com.santos.valdomiro.applistacursojava.databinding.ActivityMainBinding;
import com.santos.valdomiro.applistacursojava.model.Pessoa;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;

    Pessoa pessoa;
    PessoaController controller;

    SharedPreferences preferences;
    public static final String PREFERENCES_NAME = "pref_lista_vip";

    EditText editPrimeiroNome;
    EditText editSobrenome;
    EditText editCursoDesejado;
    EditText editTextTelefoneContato;

    Button btnLimpar;
    Button btnSalvar;
    Button btnFinalizar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        inicializarElementos();

        controller = new PessoaController();

        pessoa = new Pessoa();
        pessoa.setPrimeiroNome("Valdomiro");
        pessoa.setSobrenome("Santos");
        pessoa.setCursoDesejado("Java");
        pessoa.setTelefoneContato("19 2 9283-8274");

        Log.d("log", "onCreate: \n" +
                pessoa.getPrimeiroNome() + "\n" +
                pessoa.getSobrenome() + "\n" +
                pessoa.getCursoDesejado() + "\n" +
                pessoa.getTelefoneContato());

        editPrimeiroNome.setText(pessoa.getPrimeiroNome());
        editSobrenome.setText(pessoa.getSobrenome());
        editCursoDesejado.setText(pessoa.getCursoDesejado());
        editTextTelefoneContato.setText(pessoa.getTelefoneContato());

        salvarNoSharedPreferences(pessoa);

        btnLimpar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                limparCampos();
            }
        });

        btnFinalizar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });

        btnSalvar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                criaNovaPessoa();
            }
        });

    }

    private void criaNovaPessoa() {
        Pessoa pessoa2 = new Pessoa();

        pessoa2.setPrimeiroNome(editPrimeiroNome.getText().toString());
        pessoa2.setSobrenome(editSobrenome.getText().toString());
        pessoa2.setCursoDesejado(editCursoDesejado.getText().toString());
        pessoa2.setTelefoneContato(editTextTelefoneContato.getText().toString());

        Log.d("log", "onCreate: \n" +
                pessoa2.getPrimeiroNome() + "\n" +
                pessoa2.getSobrenome() + "\n" +
                pessoa2.getCursoDesejado() + "\n" +
                pessoa2.getTelefoneContato());
    }

    private void inicializarElementos() {
        editPrimeiroNome = binding.editPrimeiroNome;
        editSobrenome = binding.editSobrenome;
        editCursoDesejado = binding.editCursoDesejado;
        editTextTelefoneContato = binding.editContato;

        btnFinalizar = binding.btnFinalizar;
        btnSalvar = binding.btnSalvar;
        btnLimpar = binding.btnLimpar;
    }

    private void limparCampos() {
        editPrimeiroNome.setText("");
        editSobrenome.setText("");
        editCursoDesejado.setText("");
        editTextTelefoneContato.setText("");

    }

    private void salvarNoSharedPreferences(Pessoa pessoa) {
        preferences = getSharedPreferences(PREFERENCES_NAME, 0);
        @SuppressLint("CommitPrefEdits") SharedPreferences.Editor listaVip = preferences.edit();

        listaVip.putString("primeiro_nome", pessoa.getPrimeiroNome());
        listaVip.putString("sobrenome", pessoa.getSobrenome());
        listaVip.putString("curso_desejado", pessoa.getCursoDesejado());
        listaVip.putString("tel_contato", pessoa.getTelefoneContato());

        listaVip.apply();

    }

    private void recuperarDoSharedPreferences() {

    }
}

File: AppListaCurso/app/src/main/java/com/santos/valdomiro/applistacurso/view/MainActivity.kt
package com.santos.valdomiro.applistacurso.view

import android.annotation.SuppressLint
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.EditText
import android.widget.Spinner
import com.santos.valdomiro.applistacurso.controller.CursoController
import com.santos.valdomiro.applistacurso.controller.PessoaController
import com.santos.valdomiro.applistacurso.databinding.ActivityMainBinding
import com.santos.valdomiro.applistacurso.databinding.ActivitySpinnerBinding
import com.santos.valdomiro.applistacurso.model.Curso
import com.santos.valdomiro.applistacurso.model.Pessoa

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivitySpinnerBinding

    private lateinit var spinner: Spinner

    private lateinit var editPrimeiroNome: EditText
    private lateinit var editSobrenome: EditText
    private lateinit var editCursoDesejado: EditText
    private lateinit var editTelefoneContato: EditText

    private lateinit var btnLimpar: Button
    private lateinit var btnSalvar: Button
    private lateinit var btnFinalizar: Button

    private lateinit var pessoa : Pessoa

    private lateinit var controller: PessoaController
    private lateinit var cursoController: CursoController

    private lateinit var listaDeCursos: List<Curso>
    private lateinit var nomesDosCursos: List<String>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
//        binding = ActivityMainBinding.inflate(layoutInflater)
        binding = ActivitySpinnerBinding.inflate(layoutInflater)
        setContentView(binding.root)

        inicializarComponentes()

        controller = PessoaController(this)
        cursoController = CursoController()
        listaDeCursos = cursoController.listarCursos()



        nomesDosCursos = cursoController.dadosParaSpinner()

        // Adapter
        val adapter: ArrayAdapter<String> = ArrayAdapter(
            this,
            android.R.layout.simple_list_item_1,
            cursoController.dadosParaSpinner()
        )

        // Layout
        adapter.setDropDownViewResource(android.R.layout.simple_list_item_1)

        // Injetar o Adapter ao Spinner - A lista sera gerada
        spinner.adapter = adapter




        for (curso in listaDeCursos) {
            Log.d("log", "onCreate: ${curso.nomeCurso}")
        }

        for (nome in cursoController.dadosParaSpinner()) {
            Log.d("log", "nomesDeCursos: $nome")
        }

        for (nome in nomesDosCursos) {
            Log.d("log", "nomesDeCursos_outra_vez: $nome")
        }

        btnSalvar.setOnClickListener {
            controller.salvar(criarNovaPessoa())
        }

        btnLimpar.setOnClickListener {
            limparCamposESharedPreferences()
        }

        btnFinalizar.setOnClickListener {
            finish()
        }
    }

    private fun limparCamposESharedPreferences() {
        editPrimeiroNome.setText("")
        editSobrenome.setText("")
        editCursoDesejado.setText("")
        editTelefoneContato.setText("")

        controller.limpar()

    }

    private fun criarNovaPessoa(): Pessoa {
        val primeiroNome = binding.editPrimeiroNome.text.toString()
        val sobrenome = binding.editSobrenome.text.toString()
        val cursoDesejado = binding.editCursoDesejado.text.toString()
        val telefoneContato = binding.editContato.text.toString()

        pessoa = Pessoa(primeiroNome, sobrenome, cursoDesejado, telefoneContato)

        Log.d("log", "criarNovaPessoa: \n" +
                "${pessoa.primeiroNome} \n" +
                "${pessoa.sobrenome} \n" +
                "${pessoa.cursoDesejado} \n" +
                "${pessoa.telefoneContato} ")

        return pessoa
    }

    private fun inicializarComponentes() {
        editPrimeiroNome = binding.editPrimeiroNome
        editSobrenome = binding.editSobrenome
        editCursoDesejado = binding.editCursoDesejado
        editTelefoneContato = binding.editContato

        btnLimpar = binding.btnLimpar
        btnSalvar = binding.btnSalvar
        btnFinalizar = binding.btnFinalizar

        spinner = binding.spinner
    }

}